apps/documents/views.py

```diff
@@ -72,31 +73,54 @@ def update(self, request, *args, **kwargs):
         LOG.debug(f'Updating document {instance.id} with new details.')
         log_metric('transmission.info', tags={'method': 'documents.update', 'module': __name__})
 
-        serializer = DocumentRetrieveSerializer(instance, data=request.data, partial=partial,
-                                                context={'auth': request.auth})
+        serializer = self.serializer_class(instance, data=request.data, partial=partial, context={'auth': request.auth})
```

Comment:
I think the DRF convention is to use self.get_serializer() instead of self.serializer_class